File: Domain/Aggregates/WorkoutPlan.cs
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Common;
using Domain.Entities;
using Domain.Models;

namespace Domain.Aggregates
{
    [Table("WorkoutPlans")]
    public class WorkoutPlan : AggregateRoot<int>
    {
        public int CreatedById { get; set; }
        [ForeignKey("CreatedById")]
        public virtual User? CreatedBy { get; set; }
        public virtual ICollection<User> UsedBy { get; set; } = [];
        public string? Name { get; set; }
        public string? Description { get; set; }
        public WorkoutPlanCategory Category { get; set; }
        public bool IsPublic { get; set; }
        public virtual ICollection<WorkoutTemplate> WorkoutTemplates { get; set; } = [];
    }
}

File: UI/src/app/core/models/WorkoutPlan.ts
import { Exercise } from './Exercise';
import { WorkoutPlanCategory } from './Enums';
import { User } from './User';

export interface WorkoutPlan {
  id: number;
  createdById: number;
  usedBy: User[];
  name: string | null;
  description: string | null;
  category: WorkoutPlanCategory;
  categoryLabel: string;
  isPublic: boolean;
  workoutTemplates: WorkoutTemplate[];
}

export interface WorkoutTemplate {
  order: number;
  name: string | null;
  exercises: WorkoutTemplateExercise[];
}

export interface WorkoutTemplateExercise {
  order: number;
  exercise: Exercise;
  sets: number;
  minReps: number;
  maxReps: number;
}

export interface WorkoutPlanCreate {
  name: string | null;
  description: string | null;
  category: WorkoutPlanCategory;
  isPublic: boolean;
  workoutTemplates: WorkoutTemplateCreate[];
}

export interface WorkoutTemplateCreate {
  order: number;
  name: string | null;
  exercises: WorkoutTemplateExerciseCreate[];
}

export interface WorkoutTemplateExerciseCreate {
  order: number;
  exerciseId: number;
  sets: number;
  minReps: number;
  maxReps: number;
}

export interface WorkoutPlanCategoryOption {
  value: WorkoutPlanCategory;
  name: string;
  label: string;
}
